### PR TITLE
update shock-cloud problem

### PR DESCRIPTION
**Pull request by BenWibking**
This updates the shock-cloud problem to use a new setup that:

- tracks the cloud center-of-mass frame
- refines so that the cooling length $l_{\text{cool}} = c_s t_{\text{cool}}$ is resolved with at least 10 cells
- adds derived fields for temperature, entropy, pressure, etc. to the outputs
- uses the (simplified) Navier-Stokes Characteristic Boundary Conditions for inflow/outflow on the left/right sides of the box
- adds shock-cloud problem to nightly regression testing

**This PR depends on:**
* https://github.com/quokka-astro/quokka/pull/352
* https://github.com/quokka-astro/quokka/pull/360
* https://github.com/quokka-astro/quokka/pull/363
* https://github.com/quokka-astro/quokka/pull/364
* https://github.com/quokka-astro/quokka/pull/365
* https://github.com/quokka-astro/quokka/pull/366
* https://github.com/quokka-astro/quokka/pull/367
* https://github.com/quokka-astro/quokka/pull/430
* https://github.com/quokka-astro/quokka/pull/581
* https://github.com/quokka-astro/quokka/pull/596
* https://github.com/quokka-astro/quokka/pull/597
* https://github.com/quokka-astro/quokka/pull/598
* https://github.com/quokka-astro/quokka/pull/599
* https://github.com/quokka-astro/quokka/pull/616
* https://github.com/quokka-astro/quokka/pull/645

### CODE REVIEW

**Comment by BenWibking** [comment]
There appears to be a bug in the NSCBC inflow boundary conditions. FOFC is frequently triggered at cell (0,0,0), which makes no sense unless there is a bug.

File: src/hydro_system.hpp [hunk 1]
```diff
@@ -503,6 +505,10 @@ void HydroSystem<problem_t>::PredictStep(amrex::MultiFab const &consVarOld_mf, a
 		// check if state is valid -- flag for re-do if not
 		if (!isStateValid(consVarNew[bx], i, j, k)) {
 			redoFlag[bx](i, j, k) = quokka::redoFlag::redo;
+			printf("redoFlag on cell (%d, %d, %d): \n", i, j, k); // NOLINT
+			for (int n = 0; n < nvars; ++n) {
+				printf("\t(%d, %d, %d) comp %d = %g\n", i, j, k, n, consVarNew[bx](i, j, k, n)); // NOLINT
+			}
```
Comment (BenWibking) [inline src/hydro_system.hpp]: Adding this printf causes the FOFC at cell (0,0,0) to stop happening. This makes no sense.

**Comment by BenWibking** [comment]
Need to fix unused var warnings:
```
/home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp: In member function ‘void RadhydroSimulation<problem_t>::setInitialConditionsOnGrid(quokka::grid) [with problem_t = ShockCloud]’:
/home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp:93:14: error: variable ‘tables’ set but not used [-Werror=unused-but-set-variable]
   93 |         auto tables = cloudyTables_.const_tables();
      |              ^~~~~~
In file included from /home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp:26:
/home/runner/work/quokka/quokka/src/NSCBC_inflow.hpp: In instantiation of ‘void NSCBC::setInflowX1LowerLowOrder(const amrex::IntVect&, const amrex::Array4<double>&, const amrex::GeometryData&, amrex::Real, amrex::Real, amrex::Real, amrex::Real, const amrex::GpuArray<double, HydroSystem<problem_t>::nscalars_>&) [with problem_t = ShockCloud; amrex::Real = double]’:
/home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp:189:47:   required from here
/home/runner/work/quokka/quokka/src/NSCBC_inflow.hpp:170:20: error: unused variable ‘dx’ [-Werror=unused-variable]
  170 |         const Real dx = geom.CellSize(0);
      |                    ^~
/home/runner/work/quokka/quokka/src/NSCBC_inflow.hpp:171:20: error: unused variable ‘Lx’ [-Werror=unused-variable]
  171 |         const Real Lx = geom.prob_domain.length(0);
      |                    ^~
In file included from /home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp:27:
/home/runner/work/quokka/quokka/src/NSCBC_outflow.hpp: In instantiation of ‘void NSCBC::setOutflowBoundaryLowOrder(const amrex::IntVect&, const amrex::Array4<double>&, const amrex::GeometryData&, amrex::Real) [with problem_t = ShockCloud; FluxDir DIR = FluxDir::X1; NSCBC::BoundarySide SIDE = NSCBC::BoundarySide::Upper; amrex::Real = double]’:
/home/runner/work/quokka/quokka/src/ShockCloud/cloud.cpp:197:90:   required from here
/home/runner/work/quokka/quokka/src/NSCBC_outflow.hpp:373:20: error: unused variable ‘dx’ [-Werror=unused-variable]
  373 |         const Real dx = geom.CellSize(static_cast<int>(DIR));
      |                    ^~
```

File: src/ShockCloud/cloud.cpp [hunk 2]
```diff
@@ -172,133 +152,646 @@ AMREX_GPU_DEVICE AMREX_FORCE_INLINE void AMRSimulation<ShockCloud>::setCustomBou
 												const Real /*time*/, const amrex::BCRec * /*bcr*/,
 												int /*bcomp*/, int /*orig_comp*/)
 {
-	auto [i, j, k] = iv.dim3();
+	auto [i, j, k] = iv.toArray();
```
[the remark this answers is not in the document]
Comment (BenWibking) [reply src/ShockCloud/cloud.cpp:150]: This problem only builds in 3D, so this should not be an issue.

**Comment by BenWibking** [comment]
@markkrumholz I've added this problem to the nightly regression testing. It should be all ready to review.

**Review by markkrumholz — APPROVED** [review]

**Pull request merged**